checks/apps/arbor/arbor-dev.py

```diff
@@ -0,0 +1,172 @@
+                '-DARB_GPU=cuda',
+                '-DARB_VECTORIZE=on'
+            ]
+        elif self.uarch == 'zen2':
+            self.build_system.config_opts += ['-DARB_VECTORIZE=on']
+
+        self.build_system.max_concurrency = 64
+
+        self.build_system.make_opts = ['pyarb', 'examples', 'unit']
+
+@rfm.simple_test
+class arbor_unit(rfm.RunOnlyRegressionTest):
+    descr = 'Run the arbor unit tests'
+    valid_systems = ['*']
+    valid_prog_environs = ['+arbor-dev']
+    maintainers = ['bcumming']
+    arbor_build = fixture(arbor_build, scope='environment')
+
+    @run_before('run')
+    def prepare_run(self):
+        self.executable = os.path.join(self.arbor_build.stagedir, 'build', 'bin', 'unit')
+        self.executable_opts = []
+
+    @sanity_function
+    def validate_test(self):
+        return sn.assert_found(r'PASSED', self.stdout)
+
+@rfm.simple_test
+class arbor_busyring(rfm.RunOnlyRegressionTest):
+    descr = 'arbor busyring model'
+    valid_systems = ['*']
+    valid_prog_environs = ['+arbor-dev']
+    maintainers = ['bcumming']
+    model_size = parameter(['small', 'medium'])
+
+    arbor_build = fixture(arbor_build, scope='environment')
+
+    @run_before('run')
+    def prepare_run(self):
+        self.uarch = uenv.uarch(self.current_partition)
+        self.executable = os.path.join(self.arbor_build.stagedir, 'build', 'bin', 'busyring')
+        self.executable_opts = [f'busyring-input-{self.model_size}.json']
+
+        # Instead of explicitly listing performance targets for all possible
+        # system:partition combinations, set the reference targets to those
+        # for the uarch of the current partition.
+        # * self.uarch is one of the alps arch: gh200, zen2, a100, ... or None
+        # * self.current_partition.fullname is the vcluster:partition string,
+        #   for example "daint:normal" or "todi:debug".
+        self.uarch = uenv.uarch(self.current_partition)
+        if (self.uarch is not None) and (self.uarch in arbor_references):
+            self.reference = {
+                self.current_partition.fullname:
+                    arbor_references[self.uarch]['serial'][self.model_size]
+            }
+
+    @sanity_function
+    def validate_test(self):
+        # if the meters are printed, the simulation ran to completion
+        return sn.assert_found(r'meter-total', self.stdout)
+
+    @performance_function('s')
+    def time_run(self):
+        return sn.extractsingle(r'model-run\s+(\S+)', self.stdout, 1, float)
+
+slurm_config = {
+    'gh200': {"ranks": 4, "cores": 64, "gpu": True},
+    'zen2':  {"ranks": 2, "cores": 64, "gpu": False},
+}
+
+# adapt the busyring test to check paralle MPI execution
+@rfm.simple_test
+class arbor_busyring_mpi(arbor_busyring):
+    descr = 'arbor busyring model MPI on a single node'
+    model_size = parameter(['medium'])
+
+    @run_before('run')
+    def prepare_run(self):
+        self.uarch = uenv.uarch(self.current_partition)
```

Comment:
already set in `class arbor_busyring` ?

Reply:
this overwrites the `perpare_run` function (replacing the one in `class arbor_busyring`).
So whatever was set in the base class, won't be set here.